streams/src/main/java/org/apache/kafka/streams/state/KeyValueStore.java

```diff
@@ -32,7 +33,17 @@
  * @param <V> The value type
  */
 @InterfaceStability.Unstable
-public interface KeyValueStore<K, V> extends StateStore {
+public abstract class KeyValueStore<K, V> implements StateStore {
+
+    private boolean checked = false;
+
+    public void checkKeyIsArray(K key) {
+        if (!checked && key.getClass().isArray()) {
+            throw new StreamsException("KeyValueStore cannot support array keys.");
+        } else {
+            checked = true;
+        }
+    }
```

Comment:
Is it really worth doing this? Java or Scala collection classes don't do this and they all suffer from this issue, right?